spec/index.bs

```diff
@@ -4997,6 +5000,10 @@ A {{GPUImageCopyBuffer}} contains the actual [=texture=] data placed in a [=buff
 
 ### <dfn dictionary>GPUImageCopyTexture</dfn> ### {#gpu-image-copy-texture}
 
+In an image copy operation, a {{GPUImageCopyTexture}} defines a {{GPUTexture}} and, together with
```

Comment:
Forget one thing. Do we need to make GPUTexture have |RENDERATTACHMENT| usage always or do we have an extra describe to suggest user to create such texture with |RENDERATTACHMENT| usage so they can have better performance

Reply:
Why would it have a better performance?

Reply:
IIUC: If some copy operations required the RENDER_ATTACHMENT usage to be explicitly provided, then we wouldn't have to implicitly add it and they could get better performance by omitting it.

Reply:
@kvark, if the dst texture have RENDER_ATTACHMENT, browser could use a render pipeline to handle color format conversion, color space conversion and orientation issues. Without this usage, we need an intermediate resource to store the result, do the ops and omit another copy to the dst texture. In large texture, this extra copy will affect performance.

Reply:
Thank you for clarification!
In this case, we shouldn't call it a `copy` at all, to avoid confusion, since all other copies are "dump": they just reinterpret the data. This would be called "blitXxx" perhaps, instead?
And I'd love the implementations to not have 2 different code paths for this (with and without the render attachment), so requiring `RENDER_ATTACHMENT` for this updated entry point makes sense to me now.

Reply:
"writeExternalImageToTexture"?

Requiring RENDER_ATTACHMENT seems ok to me.

Reply:
I've added this tentatively, but thinking more, actually it's kind of unfortunate. Say you're uploading a PNG texture for a glTF model. You really don't want the cost of RENDER_ATTACHMENT on that forever. You'd rather pay the up-front cost of a possible additional copy.

Reply:
Is the copy doesn't matter then the application can do the extra copy on their side. For example in @toji's mipmap generation helper, if the end texture doesn't need `RENDER_ATTACHMENT` then first all mipmaps are generated in one texture, then copied over to the final texture.

Maybe in the future we can extend to say one of `RENDER_ATTACHMENT` or `COPY_DST`?

No strong opinions about the name (as in none of copy / blit / write seem good. Copy is just the least bad). Overlay? Decal? Ew.

Reply:
Both "copy" and "write" would be inconsistent with existing copy and write methods, which are strictly copies.

Reply:
Personally I think "copy" is fine. The source data is colors rather than bytes anyway, it can be thought of as a copy of the colors rather than a copy of bytes.

Reply:
None of the "copy" semantics of native APIs think about "colors", it's all about bytes. There is no need for us to be confusing here, having a single copy operation that requires `RENDER_ATTACHMENT` while all the others do not.

Reply:
"Copying" can imply a more abstract concept than those in the native APIs. Chromium's internal graphics stack has contained a "CopySubTextureCHROMIUM" abstraction for years now - copy sub-rectangle of source to destination texture, supporting conversions, and it's never been confusing for the developers who use it. I don't think it's necessary to introduce a new "blit" name to the WebGPU spec.

Reply:
Meeting resolution: keep requiring RENDER_ATTACHMENT (but add explanation of why).

No resolution on the naming yet.

Reply:
This may be overspecializing, but since we require RENDER_ATTACHMENT, we could use `drawExternalImageToTexture`

Reply:
Editors meeting: Require *both* RENDER_ATTACHMENT and COPY_DST, and keep the "copy" name